### PR TITLE
Update index.js

### PR DESCRIPTION
**Pull request by adin234**
the idea is to allow cross domain requests to share credentials.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -4,8 +4,23 @@ module.exports = function (config) {
     config = config || {};
 
     return function (req, res, next) {
-        if (config.allowed_origins) {
-            res.setHeader('Access-Control-Allow-Origin', config.allowed_origins);
+
+        if (config.allowed_origins || config.allowed_origins_list) {
```
Comment (adin234) [inline index.js]: `config.allowed_origins_list` is an array of all origins allowed

config.js

``` javascript
module.exports = {
    allowed_origins_list : [ /.*localhost(\..*)*/, /.*\.domain.com/ ]
}
```

File: index.js [hunk 2]
```diff
@@ -4,8 +4,23 @@ module.exports = function (config) {
...
+            if (config.allowed_origins_list) {
+                for (var i=0; i<config.allowed_origins_list.length; i++) {
+                    if (req.headers.origin && req.headers.origin.match(config.allowed_origins_list[i])) {
```
Comment (adin234) [inline index.js]: matche allowed origins with the origin of the recent request

File: index.js [hunk 3]
```diff
@@ -4,8 +4,23 @@ module.exports = function (config) {
...
+                        res.setHeader('Access-Control-Allow-Credentials', config.allow_credentials);
+                        res.setHeader('Access-Control-Allow-Origin', req.headers.origin);
+                        break;
+                    }
+                }
+
+                if (i==config.allowed_origins_list.length || !req.headers.origin) {
```
Comment (adin234) [inline index.js]: `!req.headers.origin` to handle non-xhr

**Comment by ravenjohn** [comment]
suddenly forgot the JS conventions?

**Comment by adin234** [comment]
updated.

File: index.js [hunk 4]
```diff
@@ -4,8 +4,24 @@ module.exports = function (config) {
...
+                        res.setHeader('Access-Control-Allow-Credentials', config.allow_credentials);
+                        res.setHeader('Access-Control-Allow-Origin', req.headers.origin);
+                        break;
+                    }
+                }
+
+                if (i === config.allowed_origins_list.length || !req.headers.origin) {
+                    res.setHeader('Access-Control-Allow-Origin', config.allowed_origins);
+                }
+            } else {
```
Comment (ravenjohn) [inline index.js]:
``` javascript
}
else {
```

**Comment by adin234** [comment]
updated

File: index.js [hunk 5]
```diff
@@ -4,8 +4,25 @@ module.exports = function (config) {
...
+                        res.setHeader('Access-Control-Allow-Origin', req.headers.origin);
+                        break;
+                    }
+                }
+
+                if (i === config.allowed_origins_list.length || !req.headers.origin) {
+                    res.setHeader('Access-Control-Allow-Origin', config.allowed_origins);
+                }
+            }
+            else {
```
Comment (ravenjohn) [inline index.js]:
``` javascript
else if (config.allowed_origins) {
```

then remove the outer `if`

**Comment by adin234** [comment]
updated

**Pull request merged**